internal/testutil/query.go

```diff
@@ -20,25 +20,43 @@ func Query(query string, res any, opts *QueryOptions) error {
 }
 
 func QueryWithEngineConfig(query string, res any, opts *QueryOptions, cfg *engine.Config) error {
+	ctx := context.Background()
+
 	if opts == nil {
 		opts = &QueryOptions{}
 	}
 	if opts.Variables == nil {
 		opts.Variables = make(map[string]any)
 	}
-	return engine.Start(context.Background(), cfg, func(ctx engine.Context) error {
-		return ctx.Client.MakeRequest(ctx,
-			&graphql.Request{
-				Query:     query,
-				Variables: opts.Variables,
-				OpName:    opts.Operation,
-			},
-			&graphql.Response{Data: &res},
+
+	clientOpts := []dagger.ClientOpt{}
+	if cfg != nil {
+		clientOpts = append(clientOpts,
+			dagger.WithWorkdir(cfg.Workdir),
+			dagger.WithConfigPath(cfg.ConfigPath),
 		)
-	})
+		for id, path := range cfg.LocalDirs {
+			clientOpts = append(clientOpts, dagger.WithLocalDir(id, path))
+		}
+	}
+
+	c, err := dagger.Connect(ctx, clientOpts...)
+	if err != nil {
+		return err
+	}
+	defer c.Close()
+
+	return c.Do(ctx,
+		&dagger.Request{
+			Query:     query,
+			Variables: opts.Variables,
+			OpName:    opts.Operation,
+		},
+		&dagger.Response{Data: &res},
+	)
 }
 
-func ReadFile(ctx context.Context, cl graphql.Client, fsid dagger.FSID, path string) (string, error) {
+func ReadFileXX(ctx context.Context, cl graphql.Client, fsid dagger.FSID, path string) (string, error) {
```

Comment:
typo? or just cooler name? 😎

Reply:
😎 

Forgot to remove it. It was a quick check to see if the function was being used, and it's not. I'll remove it altogether